### DISCUSSION TITLE
Using mod header extention with profile.json

### DISCUSSION THREAD

**Discussion by yoelfr** [cypress-io/cypress#27290] [Questions and Help]
I need to use mod header for specific site. 
I use cypress-browser-extension-plugin to add mod header plugin and it works 
but I need a way how to add the headers to the extention  or the profile file to the mod header because now the mod header is empty 
THANKS

**Comment by portyu9** [comment]
Hi @yoelfr 

If the requirement is simply “send additional headers to this site”, I would avoid ModHeader entirely and use Cypress's network API.

Register the intercept **before** visiting the page:

```js
beforeEach(() => {
  cy.intercept(
    {
      url: 'https://example.com/**',
      middleware: true,
    },
    (req) => {
      req.headers['x-my-header'] = 'my-value'
    }
  )
})
```

That header will be applied to matching outgoing application requests.

If you only need a header on the initial document request, `cy.visit()` also supports:

```js
cy.visit({
  url: 'https://example.com',
  headers: {
    'x-my-header': 'my-value',
  },
})
```

but note that `cy.visit({ headers })` applies those headers only to the **initial visit request**, not subsequent requests made by the page.

If you genuinely need to test the ModHeader extension itself, Cypress can load unpacked extensions through `before:browser:launch`, but relying on extension profile state is much more fragile. Current Cypress also has an important limitation: normal Chrome 137+ no longer supports extension loading through this API. Chrome for Testing and Chromium are the supported alternatives for that use case.

So I would not make `profile.json` persistence the solution unless testing the extension itself is actually part of the requirement. For request headers, `cy.intercept()` is the supported and deterministic Cypress approach.

Please mark this answer as accepted if it helped, thank you.
